test/e2e/scrapeconfig_test.go

```diff
@@ -785,8 +797,38 @@ func testScrapeConfigCRDValidations(t *testing.T) {
 			},
 			expectedError: true,
 		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
```

Comment:
the bulk of the function is duplicated between all sub tests. Can we improve?

Reply:
I was pondering this too, figured I will make a first draft then improve if anyone has ideas. 
Let me see what I can do.